backend/main_page/views/signup_view.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from django.contrib.auth.models import User
+
+class SignupView(APIView):
+    def post(self, request):
+        # Extract necessary fields from headers
+        username = request.data.get("username", "")
+        email = request.data.get("email", "")
+        password = request.data.get("password", "")
+        verify_password = request.data.get("verify_password", "")
+
+        # Bare minimum field verification
+        if password != verify_password:
+            return Response({"error": "The password doesn't match"}, status=status.HTTP_400_BAD_REQUEST)
+
+        # Verify email format (example: gmail or yahoo)
+        if "gmail" not in email and "yahoo" not in email:
```

Comment:
This part could some issues up, an example for this part that would be accepted but not correct is: gmail@idontcare.com